wallet-new/server/Cardano/Wallet/API/V1/LegacyHandlers/Transactions.hs

```diff
@@ -118,6 +122,46 @@ estimateFees Payment{..} = do
     cAccountId <- migrate pmtSource
     utxo <- V0.getMoneySourceUtxo ws (V0.AccountMoneySource cAccountId)
     outputs <- V0.coinDistrToOutputs =<< mapM migrate pmtDestinations
-    fee <- V0.rewrapTxError "Cannot compute transaction fee" $
-        eitherToThrow =<< V0.runTxCreator policy (V0.computeTxFee pendingAddrs utxo outputs)
-    single <$> migrate fee
+    efee <- V0.runTxCreator policy (V0.computeTxFee pendingAddrs utxo outputs)
+    single <$> case efee of
+        Left txError ->
+            case txError of
+                V0.NotEnoughMoney coin -> do
+                    acct <- V0.getAccount cAccountId
+                    pure $ mkLowerBound acct coin
+                V0.NotEnoughAllowedMoney coin -> do
+                    acct <- V0.getAccount cAccountId
+                    pure $ mkLowerBound acct coin
+                _ ->
+                    throwM (transactionErrorToWalletError txError)
+        Right fee ->
+            migrate fee
+  where
+    mkLowerBound
+        :: V0.CAccount
+        -- ^ The account that failed to pay.
+        -> Core.Coin
+        -- ^ The amount of coins necessary to complete the transaction.
+        -> EstimatedFees
+    mkLowerBound acct amountUnder =
+        EstimatedFees (V1 (Core.mkCoin (fromIntegral feeLowerBound))) LowerBound
+      where
+        feeLowerBound =
+            totalToSpend - txnTotal
+        totalToSpend =
+            Core.getCoin amountUnder + acctAmount
```

Comment:
i am absolutely terrible at math, so someone please check me on this:

example:

Attempt to pay 90 coins
I have 100 coins in balance
I get the error back "Need 10 more coins"
The assumed total of transaction is (balance + amount needed = 110 coins)
The assumed fee is the assumed total minus the amount we try to pay (20 coins).

Reply:
Let's translate your example into code, but let's assume something different: _we need to pay 100 but we only have 90 in the balance_ . What you posted in your example is a perfectly valid transaction as I do have enough money to pay for it -- what am I missing? 😀 . We have:

```haskell
let amountUnder = 10 -- let's assume I have 90 coins, I need 10 more to fullfill 100.
let fee= EstimatedFees (V1 (Core.mkCoin (fromIntegral feeLowerBound))) LowerBound
let feeLowerBound = totalToSpend - txnTotal
let totalToSpend = Core.getCoin amountUnder + acctAmount
let txnTotal = sum (fmap (Core.getCoin . unV1 . pdAmount) pmtDestinations)
let acctAmount = sum (fmap (Core.getCoin . Txp.txOutValue . Txp.toaOut) utxo)
```

reducing:

```
let amountUnder = 10
let fee= EstimatedFees (V1 (Core.mkCoin (fromIntegral feeLowerBound))) LowerBound
let feeLowerBound = totalToSpend - txnTotal
let totalToSpend = 10 + acctAmount
let txnTotal = 100 -- we need to pay 100 coins to all the various destinations.
let acctAmount = 90 -- we have 90 coins in our "bank"

=> 

let amountUnder = 10
let fee= EstimatedFees (V1 (Core.mkCoin (fromIntegral feeLowerBound))) LowerBound
let feeLowerBound = 100 - 100
let totalToSpend = 10 + 90 = 100
let txnTotal = 100
let acctAmount = 90
```

This shows that essentially `totalToSpend == txnTotal` and that your fee calculation will always be 0 if you assume that you have _exactly_ the required amount of money to fullfil the contract. But of course a `Fee` of 0 is not correct, and I recommend not going into that rabbit hole. After reading [our documentation](https://cardanodocs.com/cardano/transaction-fees/) here is what I would do: I would reuse the formula in the docs:

```
a + b × size
```

Where `a` and `b` are constants, and `size` is the only unknown. Of course, we cannot determine `size` _apriori_ without having a valid Tx at hand, so I would suggest reusing the `txToLinearFee` function to do so:

``` haskell
txToLinearFee
    :: MonadError TxError m
    => TxSizeLinear -> TxAux -> m TxFee
txToLinearFee linearPolicy =
    fixedToFee .
    calculateTxSizeLinear linearPolicy .
    biSize @TxAux
```

Of course do don't have all the inputs required to generate a valid fee, but this shouldn't matter as this is a lower bound -- I would give a shot to `createFakeTxFromRawTx` to see if we can create a fake transaction with essentially all the inputs we own (and the outputs we need to pay for) -- granted, it won't be _valid_, but that's is the closest approximation in terms of bytes we have at the moment.

What do you think?

Reply:
I don't think that approach buys us anything -- the `TxError` type that we get back is coming from one of these functions anyway, and if we get another "Need more money pls" error then it doesn't do much for us :)

I'll write some tests for the math and factor it out into it's own function.

Reply:
My point is that if we are not following a specification in order to compute even a lower bound, what is the value of reporting it 😀 ? Imho the right way to go would still to apply the formula, otherwise we are effectively using two different fees calculations in two different circumstances, no? 

To be sure to be on the same page, I'm not advocating to modify the internal or to go for any weird logic, but just to swap the code inside `mkLowerBound` to reuse the same formula we use for the accurate calculation, as that will likely give more realistic calculations 😉 

Happy to have a voice chat about that -- in the meantime you are asleep will ask also other folks who knows better than me when it comes to fee calculation, so see if we can find the sweet spot between the two approaches :)